### PR TITLE
Add a basic-ish debounce implementation, exported as a helper

### PR DESCRIPTION
**Pull request by stewart**
Similar, but not quite, the same as throttling function invocations.

> **debounce (v)**: To remove the small ripple of current that forms when a mechanical switch is pushed in an electrical circuit and makes a series of short contacts.

The specs should paint a better picture of how this works and how it can be used.

### CODE REVIEW

File: lib/utils/helpers.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // A collection of useful helper functions, used internally but not exported
 // with the rest of Cylon. 
-"use strict";
+
+// jshint strict:false
```
Comment (stewart) [inline lib/utils/helpers.js]: Got rid of this because of the new debounce method, `this` inside a function is a strict mode violation.

**Comment by stewart** [comment]
Alternatively, instead of more code for us to maintain, we could just lean on packages like https://www.npmjs.com/package/debounce

**Comment by stewart** [comment]
Since it's not going to be used anywhere in core, I'll close this PR. If a user's looking for a debounce implementation, or if we want to use a similar utility in a module, we can just use the existing NPM packages.

**Comment by bartmichu** [comment]
Hey this is really needed, I don't know how you are able to work with buttons without debouncing.

**Pull request closed without merging**